src/integrations/integrations.test.ts

```diff
@@ -0,0 +1,343 @@
+import {
+  expect,
+} from 'chai';
+
+import {
+    GraphQLSchema,
+    GraphQLObjectType,
+    GraphQLString,
+} from 'graphql';
+
+// tslint:disable-next-line
+const request = require('supertest-as-promised');
+
+import ApolloOptions from './apolloOptions';
+
+import * as GraphiQL from '../modules/renderGraphiQL';
+import { OperationStore } from '../modules/operationStore';
+
+const QueryType = new GraphQLObjectType({
+    name: 'QueryType',
+    fields: {
+        testString: {
+            type: GraphQLString,
+            resolve() {
+                return 'it works';
+            },
+        },
+        testArgument: {
+            type: GraphQLString,
+            args: { echo: { type: GraphQLString } },
+            resolve(root, { echo }) {
+                return `hello ${echo}`;
+            },
+        },
+    },
+});
+
+const MutationType = new GraphQLObjectType({
+    name: 'MutationType',
+    fields: {
+        testMutation: {
+            type: GraphQLString,
+            args: { echo: { type: GraphQLString } },
+            resolve(root, { echo }) {
+                return `not really a mutation, but who cares: ${echo}`;
+            },
+        },
+    },
+});
+
+export const Schema = new GraphQLSchema({
+    query: QueryType,
+    mutation: MutationType,
+});
+
+export interface CreateAppOptions {
+  excludeParser?: boolean;
+  apolloOptions?: ApolloOptions | {(): ApolloOptions | Promise<{}>};
+  graphiqlOptions?: GraphiQL.GraphiQLData;
+}
+
+export interface CreateAppFunc {
+    (options?: CreateAppOptions): void;
+}
+
+export default (createApp: CreateAppFunc) => {
```

Comment:
I would add a string parameter here to describe the tests, for example "Express" or "HAPI"